### PR TITLE
test: Set devices and enable host firewall in kube-proxy CI

### LINKED ISSUE #11972: Setting devices without NodePort breaks multi-node connectivity
Setting `--device` with the private iface seems to break multi-node connectivity in multiple scenarios. For example, commit 7dc74091e8b79b606268539754a31f41e5e3f96e implements that change in CI and https://jenkins.cilium.io/job/Cilium-PR-K8s-newest-kernel-4.9/654/testReport/ has the corresponding results. The following tests are failing:
```
Suite-k8s-1.18.K8sPolicyTest Basic Test Allows traffic with k8s default-allow egress policy
Suite-k8s-1.18.K8sPolicyTest Basic Test Validate to-entities policies Validate toEntities All
Suite-k8s-1.18.K8sPolicyTest Basic Test Validate to-entities policies Validate toEntities World
Suite-k8s-1.18.K8sPolicyTest Basic Test Validate to-entities policies Validate toEntities Cluster
Suite-k8s-1.18.K8sPolicyTest Basic Test Traffic redirections to proxy Tests DNS proxy visibility without policy
Suite-k8s-1.18.K8sPolicyTest  Multi-node policy test validates fromEntities policies with remote-node  identity disabled Allows from all hosts with cnp fromEntities host  policy
Suite-k8s-1.18.K8sPolicyTest  Multi-node policy test validates fromEntities policies with remote-node  identity enabled Validates fromEntities remote-node policy
Suite-k8s-1.18.K8sIstioTest Istio Bookinfo Demo Tests bookinfo inter-service connectivity
Suite-k8s-1.18.K8sHealthTest checks cilium-health status between nodes
Suite-k8s-1.18.K8sFQDNTest Validate that multiple specs are working correctly
Suite-k8s-1.18.K8sConformance Portmap Chaining Check connectivity-check compliance with portmap chaining
Suite-k8s-1.18.K8sConformance Portmap Chaining Check one node connectivity-check compliance with portmap chaining
Suite-k8s-1.18.K8sDatapathConfig MonitorAggregation Checks that monitor aggregation restricts notifications
Suite-k8s-1.18.K8sDatapathConfig MonitorAggregation Checks that monitor aggregation flags send notifications
Suite-k8s-1.18.K8sDatapathConfig Encapsulation Check connectivity with sockops and VXLAN encapsulation
Suite-k8s-1.18.K8sDatapathConfig Encapsulation Check connectivity with VXLAN encapsulation
Suite-k8s-1.18.K8sDatapathConfig Encapsulation Check connectivity with Geneve encapsulation
Suite-k8s-1.18.K8sDatapathConfig AutoDirectNodeRoutes Check connectivity with automatic direct nodes routes
Suite-k8s-1.18.K8sDatapathConfig AutoDirectNodeRoutes Check direct connectivity with per endpoint routes
Suite-k8s-1.18.K8sDatapathConfig AutoDirectNodeRoutes Check connectivity with sockops and direct routing
Suite-k8s-1.18.K8sDatapathConfig  Transparent encryption DirectRouting Check connectivity with  transparent encryption and direct routing
Suite-k8s-1.18.K8sDatapathConfig IPv4Only Check connectivity with IPv6 disabled
Suite-k8s-1.18.K8sDatapathConfig Etcd Check connectivity
Suite-k8s-1.18.K8sHubbleTest Hubble Observe Test L3/L4 Flow with hubble-relay
Suite-k8s-1.18.K8sHubbleTest Hubble Observe Test L7 Flow with hubble-relay
Suite-k8s-1.18.K8sServicesTest Checks service across nodes Checks ClusterIP Connectivity
Suite-k8s-1.18.K8sServicesTest Checks service across nodes Tests NodePort (kube-proxy)
Suite-k8s-1.18.K8sServicesTest Checks service across nodes Tests NodePort (kube-proxy) with externalTrafficPolicy=Local
Suite-k8s-1.18.K8sServicesTest Checks service across nodes with L4 policy Tests NodePort with L4 Policy
Suite-k8s-1.18.K8sServicesTest Checks service across nodes with L7 policy Tests NodePort with L7 Policy
Suite-k8s-1.18.K8sKafkaPolicyTest Kafka Policy Tests KafkaPolicies
Suite-k8s-1.18.K8sChaosTest Connectivity demo application Endpoint can still connect while Cilium is not running
Suite-k8s-1.18.K8sDemosTest Tests Star Wars Demo
```
I am able to reproduce locally, although the connectivity checks work fine with `--device=enp0s3,enp0s8` in the multi-node dev. Vagrant setup.

I'm setting as `priority/release-blocker` because it is needed for the host firewall.

/cc @brb 

Related: #11969, #11799.

### PR DESCRIPTION
**Pull request by pchaigno**
This pull request sets devices and enables the host firewall in kube-proxy CIs.
See commit messages for details.

Fixes: #11972

### CODE REVIEW

**Comment by pchaigno** [comment]
retest-4.9

**Comment by pchaigno** [comment]
retest-4.9

**Comment by pchaigno** [comment]
retest-4.9

**Comment by pchaigno** [comment]
retest-4.9

**Comment by pchaigno** [comment]
retest-4.9

**Comment by pchaigno** [comment]
I looked a bit into the kube-dns issue. Apparently, in direct routing mode, DNS replies are coming from the wrong address:
```
$ dig kubernetes.default.svc.cluster.local @10.0.1.157
;; reply from unexpected source: 192.168.36.12#53, expected 10.0.1.157#53
;; reply from unexpected source: 192.168.36.12#53, expected 10.0.1.157#53
```
The query reaches the second node with kube-dns as expected, but then the answer leaves the second node with source IP address `192.168.36.12` instead of the `10.0.1.157` expected by the client. It looks like we are either (1) SNATing at second node when we shouldn't or (2) missing a reverse SNATing.

**Comment by brb** [comment]
We should either disable masquerading for those tests `global.masquerade=false`, or if the pod cidr allocation is known, then set `global.nativeRountingCIDR`. The latter will change the masquerade rule from `-A CILIUM_POST_nat -s 10.0.1.0/24 ! -d 10.0.1.0/24 ! -o cilium_+ -m comment --comment "cilium masquerade non-cluster" -j MASQUERADE` to `A CILIUM_POST_nat -s 10.0.1.0/24 ! -d 10.0.0.0/16 ! -o cilium_+ -m comment --comment "cilium masquerade non-cluster" -j MASQUERADE` which will prevent from the unnecessary masquerading which makes the rev-DNAT xlation impossible.

**Comment by pchaigno** [comment]
test-focus K8sDatapathConfig

**Comment by pchaigno** [comment]
test-focus K8sDatapathConfig

**Comment by pchaigno** [comment]
test-focus K8sDatapathConfig Transparent encryption DirectRouting Check connectivity with transparent encryption and direct routing

**Comment by pchaigno** [comment]
When running with `global.nativeRoutingCIDR=10.0.0.0/16`, the transparent encryption has the following warning:
```
level=debug msg="IPsec enable succeeded" local-ip=10.0.0.176/24 reason="local IPv4" remote-ip=0.0.0.0/0 spi=6 subsys=linux-datapath
level=debug msg="IPsec enable succeeded" local-ip="fd00::41/120" reason="local IPv6" remote-ip="<nil>" spi=6 subsys=linux-datapath
level=debug msg=insertNeighbor HardwareAddr="08:00:27:1b:94:ef" Interface=enp0s8 LinkIndex=3 ipAddr=192.168.36.11 reason="insertNeighbor NeighSet" subsys=linux-datapath
level=debug msg="Updating direct route" allocCIDR=10.0.1.0/24 ipAddr=192.168.36.11 subsys=linux-datapath
level=debug msg="Updating direct route" allocCIDR="fd00::100/120" ipAddr="<nil>" subsys=linux-datapath
level=warning msg="Unable to install direct node route {Ifindex: 0 Dst: fd00::100/120 Src: <nil> Gw: <nil> Flags: [] Table: 0}" error="unable to lookup route for node <nil>: numerical result out of range" subsys=linux-datapath
```
Running with `global.masquerade=false` doesn't seem to have the same issue.

**Comment by pchaigno** [comment]
retest-4.9

**Comment by pchaigno** [comment]
retest-4.9

**Comment by pchaigno** [comment]
test-me-please

**Comment by pchaigno** [comment]
test-gke

**Comment by pchaigno** [comment]
test-focus K8sDatapathConfig AutoDirectNodeRoutes Check connectivity with automatic direct nodes routes

**Comment by pchaigno** [comment]
test-me-please

**Comment by pchaigno** [comment]
test-gke

**Comment by pchaigno** [comment]
test-me-please

**Comment by pchaigno** [comment]
test-gke

**Comment by pchaigno** [comment]
test-focus K8sDatapathConfig Encapsulation Check connectivity with Geneve encapsulation

**Comment by pchaigno** [comment]
test-me-please

**Comment by pchaigno** [comment]
test-gke

**Comment by pchaigno** [comment]
retest-runtime

**Review by aanm — APPROVED** [review]

File: test/k8sT/DatapathConfiguration.go [hunk 1]
```diff
@@ -308,10 +308,14 @@ var _ = Describe("K8sDatapathConfig", func() {
 		})
 
 		It("Check connectivity with automatic direct nodes routes", func() {
-			deploymentManager.DeployCilium(map[string]string{
+			options := map[string]string{
 				"global.tunnel":               "disabled",
 				"global.autoDirectNodeRoutes": "true",
-			}, DeployCiliumOptionsAndDNS)
+			}
+			if helpers.RunsWithKubeProxy() {
```
Comment (brb) [inline test/k8sT/DatapathConfiguration.go:316]: Maybe a small comment explaining why (1) when we run w/ kube-proxy, (2) we disable masquerade for this particular test case.

**Review by brb — APPROVED** [review]
LGTM, just one suggestion to add a comment.

File: test/k8sT/DatapathConfiguration.go (hunk 1 above)
Comment (pchaigno) [reply test/k8sT/DatapathConfiguration.go:316]: @brb TBH, I'm not sure I understand why this change is necessary. It is supposed to work with masquerade enabled, isn't it? Is there an issue opened somewhere to track this?

**Pull request merged**